### PR TITLE
[IP2-70] Support more RDBMS in setup

### PR DESCRIPTION
**Pull request by 749**
Pull Request Checklist

  * [x] My code follows the code formatting guidelines.
  * [x] I have an issue ID for this pull request. 
  * [x] I selected the corresponding branch.
  * [x] I have rebased my changes on top of the corresponding branch.
  
Issue Type (Please check one or more)

  * [ ] Bugfix
  * [x] Improvement of an existing Feature 
  * [ ] New Feature

[Issue](https://development.invoiceplane.com/browse/IP2-70)

image

image

### CODE REVIEW

File: app/Modules/Setup/Controllers/SetupController.php [hunk 1]
```diff
@@ -75,7 +75,26 @@ public function prerequisites()
...
+        $drivers = array();
+        foreach (\PDO::getAvailableDrivers() as $driver) {
+            // check if driver is supported by laravel
+            if(config('database.connections.' . $driver) != null) {
+                $drivers[$driver] = $driver;
+            }
+        }
+
+        // prepare the database name if we are using the sqlite driver
+        $current_settings = config('database.connections.mysql');
```
Comment (Seb33300) [inline app/Modules/Setup/Controllers/SetupController.php]: The comment just above refer to sqlite but you are getting the mysql config `database.connections.mysql`?
Is it normal?

**Comment by Seb33300** [comment]
I think SQLite database file should be stored in the `storage` folder and not in the `database` folder.

And maybe we should not let the choice to the user to choose the database filename.
I don't really see any advantage for the user to choose the name of the file, and this make the installation process more complex since the user may not understand the purpose of the field.

File: app/Modules/Setup/Controllers/SetupController.php (hunk 1 above)
Comment (749) [reply app/Modules/Setup/Controllers/SetupController.php]: The configuration values are replicated across the drivers. I just use the MySQL one as it has all the fields that the user can edit. See the config/database.php file. I will add a comment to explain.

**Comment by 749** [comment]
I just used the default path that laravel uses for sqlite, it is easy to change though. Just let me know which version you would like.

**Comment by 749** [comment]
Added a message in case no PDO drivers are available:
image

**Comment by 749** [comment]
Just noticed: the database backup uses Mysqldump, that will not work sqlite or the other RDBMS.
I suggest we take a look at [https://github.com/spatie/db-dumper](https://github.com/spatie/db-dumper)

**Pull request closed without merging**